src/llvm/orc/jit_dylib.cr

```diff
@@ -10,7 +10,8 @@ class LLVM::Orc::JITDylib
   end
 
   def link_symbols_from_current_process : Nil
-    LLVM.assert LibLLVM.orc_create_dynamic_library_search_generator_for_process(out dg, 0, nil, nil)
+    prefix = {% if flag?(:darwin) %} '_'.ord.to_u8! {% else %} 0_u8 {% end %}
```

Comment:
**question:** What's the purpose of this prefix? Should there be a code comment about it?

Reply:
It is here: https://github.com/llvm/llvm-project/blob/c4206f1ff1d946c495b74738eba9303dafb17e18/llvm/include/llvm-c/Orc.h#L1005-L1008 and also here: https://llvm.org/docs/ORCv2.html#how-to-manage-symbol-strings

The latter suggests that C++'s `LLVM::DataLayout::getGlobalPrefix` should be used for this, which maps to `LLVMOrcLLJITGetGlobalPrefix`:

https://github.com/llvm/llvm-project/blob/c4206f1ff1d946c495b74738eba9303dafb17e18/llvm/include/llvm-c/LLJIT.h#L146-L149